controller/golden_files_test.go

```diff
@@ -66,48 +93,177 @@ func testableCompareWithGolden(update bool, goldenFile string, actualObj interfa
+	if err != nil {
+		return nil, errs.Wrapf(err, "failed to compile UUID regex pattern %s", pattern)
+	}
+	uniqIDs := map[uuid.UUID]struct{}{}
+	var res []uuid.UUID
+	for _, uuidStr := range uuidRegexp.FindAllString(str, -1) {
+		ID, err := uuid.FromString(uuidStr)
+		if err != nil {
+			return nil, errs.Wrapf(err, "failed to parse UUID %s", uuidStr)
+		}
+		_, alreadyInMap := uniqIDs[ID]
+		if !alreadyInMap {
+			uniqIDs[ID] = struct{}{}
+			// append to array
+			res = append(res, ID)
+		}
+	}
+	return res, nil
+}
+
+// replaceUUIDs finds all UUIDs in the given string and replaces them with
+// "00000000-0000-0000-0000-000000000001,
+// "00000000-0000-0000-0000-000000000002", ...,
+// "00000000-0000-0000-0000-00000000000N"
+func replaceUUIDs(str string) (string, error) {
+	replacementPattern := "00000000-0000-0000-0000-%012d"
+	ids, err := findUUIDs(str)
+	if err != nil {
+		return "", errs.Wrapf(err, "failed to find UUIDs in string %s", str)
+	}
+	newStr := str
+	for idx, id := range ids {
+		newStr = strings.Replace(newStr, id.String(), fmt.Sprintf(replacementPattern, idx+1), -1)
+	}
+	return newStr, nil
+}
+
+const testInputStr = `
+{
+	"data": {
+		"attributes": {
+		"createdAt": "0001-01-01T00:00:00Z",
+		"last_used_workspace": "my-last-used-workspace",
+		"type": "git",
+		"url": "https://github.com/fabric8-services/fabric8-wit.git"
+		},
+		"id": "d7a282f6-1c10-459e-bb44-55a1a6d48bdd",
+		"links": {
+		"edit": "http:///api/codebases/d7a282f6-1c10-459e-bb44-55a1a6d48bdd/edit",
+		"related": "http:///api/codebases/d7a282f6-1c10-459e-bb44-55a1a6d48bdd",
+		"self": "http:///api/codebases/d7a282f6-1c10-459e-bb44-55a1a6d48bdd"
+		},
+		"relationships": {
+		"space": {
+			"data": {
+			"id": "a8bee527-12d2-4aff-9823-3511c1c8e6b9",
+			"type": "spaces"
+			},
+			"links": {
+			"related": "http:///api/spaces/a8bee527-12d2-4aff-9823-3511c1c8e6b9",
+			"self": "http:///api/spaces/a8bee527-12d2-4aff-9823-3511c1c8e6b9"
+			}
+		}
+		},
+		"type": "codebases"
+	}
+}`
+
+const testOutputStr = `
+{
+	"data": {
+		"attributes": {
+		"createdAt": "0001-01-01T00:00:00Z",
+		"last_used_workspace": "my-last-used-workspace",
+		"type": "git",
+		"url": "https://github.com/fabric8-services/fabric8-wit.git"
+		},
+		"id": "00000000-0000-0000-0000-000000000001",
+		"links": {
+		"edit": "http:///api/codebases/00000000-0000-0000-0000-000000000001/edit",
+		"related": "http:///api/codebases/00000000-0000-0000-0000-000000000001",
+		"self": "http:///api/codebases/00000000-0000-0000-0000-000000000001"
+		},
+		"relationships": {
+		"space": {
+			"data": {
+			"id": "00000000-0000-0000-0000-000000000002",
+			"type": "spaces"
+			},
+			"links": {
+			"related": "http:///api/spaces/00000000-0000-0000-0000-000000000002",
+			"self": "http:///api/spaces/00000000-0000-0000-0000-000000000002"
+			}
+		}
+		},
+		"type": "codebases"
+	}
+}`
+
+func TestFindUUIDs(t *testing.T) {
+	t.Parallel()
+	t.Run("find UUIDs", func(t *testing.T) {
+		t.Parallel()
```

Comment:
just curious: why do we need `t.Parallel()` here and above as well ?

Reply:
Parallel works nested if I remember correctly. If you want your sub-tests to run in parallel you would have to run `t.Parallel()` in them. Then they are allowed to run in parallel with each other. If you want this group of sub-tests to run in parallel with other parallel tests you need to run `t.Parallel()` on a level higher.